tests/integration/test_schema_protobuf.py

```diff
@@ -104,3 +104,142 @@ async def test_protobuf_schema_compatibility(registry_async_client: Client, trai
     )
     assert res.status_code == 200
     assert "id" in res.json()
+
+
+async def test_protobuf_schema_references(registry_async_client: Client) -> None:
+    customer_schema = """
+            |syntax = "proto3";
+            |package a1;
+            |message Customer {
+            |        string name = 1;
+            |        int32 code = 2;
+            |}
+            |"""
+
+    customer_schema = trim_margin(customer_schema)
+    res = await registry_async_client.post(
+        "subjects/customer/versions", json={"schemaType": "PROTOBUF", "schema": customer_schema}
+    )
+    assert res.status_code == 200
+    assert "id" in res.json()
+    original_schema = """
+            |syntax = "proto3";
+            |package a1;
+            |import "Customer.proto";
+            |message TestMessage {
+            |    enum Enum {
+            |       HIGH = 0;
+            |       MIDDLE = 1;
+            |       LOW = 2;
+            |    }
+            |    message Value {
+            |        message Label{
+            |              int32 Id = 1;
+            |              string name = 2;
+            |        }
+            |        Customer customer = 1;
+            |        int32 x = 2;
+            |    }
+            |    string test = 1;
+            |    .a1.TestMessage.Value val = 2;
+            |    oneof page_info {
+            |      option (my_option) = true;
+            |      int32 page_number = 3;
+            |      int32 result_per_page = 4;
+            |    }
+            |}
+            |"""
+
+    original_schema = trim_margin(original_schema)
+    references = [{"name": "Customer.proto", "subject": "customer", "version": 1}]
+    res = await registry_async_client.post(
+        "subjects/test_schema/versions",
+        json={"schemaType": "PROTOBUF", "schema": original_schema, "references": references},
+    )
+    assert res.status_code == 200
+    assert "id" in res.json()
+    res = await registry_async_client.get("subjects/customer/versions/latest/referencedby", json={})
+    assert res.status_code == 200
+    myjson = res.json()
+    referents = [2]
+    assert not any(x != y for x, y in zip(myjson, referents))
+
+    res = await registry_async_client.delete("subjects/customer/versions/1")
+    assert res.status_code == 404
+    match_msg = "Subject 'customer' Version 1 cannot be not be deleted because it is referenced by schemas with ids:[2]"
+
```

Comment:
Add also the case where the referencing schema is deleted and this schema can be deleted.

Reply:
ok